x/tokenregistry/keeper/msg_server.go

```diff
@@ -47,6 +49,28 @@ func (m msgServer) Deregister(ctx context.Context, req *types.MsgDeregister) (
 
 }
 
+func (srv msgServer) TokenMetadataAdd(goCtx context.Context, msg *types.TokenMetadataAddRequest) (*types.TokenMetadataAddResponse, error) {
```

Comment:
Where do we make sure that only admins can call this?

Reply:
it is checked in  AddIBCTokenMetadata
func (k keeper) AddIBCTokenMetadata(ctx sdk.Context, metadata types.TokenMetadata, cosmosSender sdk.AccAddress) string {
	.........
	if !k.IsAdminAccount(ctx, cosmosSender) {